diff -uNr linux-6.1.65_orig/drivers/mtd/ubi/build.c linux-6.1.65/drivers/mtd/ubi/build.c
--- linux-6.1.65_orig/drivers/mtd/ubi/build.c	2023-12-03 14:32:13.000000000 +0800
+++ linux-6.1.65/drivers/mtd/ubi/build.c	2023-12-05 08:56:59.438110239 +0800
@@ -879,20 +879,6 @@
 		return -EINVAL;
 	}
 
-	/*
-	 * Both UBI and UBIFS have been designed for SLC NAND and NOR flashes.
-	 * MLC NAND is different and needs special care, otherwise UBI or UBIFS
-	 * will die soon and you will lose all your data.
-	 * Relax this rule if the partition we're attaching to operates in SLC
-	 * mode.
-	 */
-	if (mtd->type == MTD_MLCNANDFLASH &&
-	    !(mtd->flags & MTD_SLC_ON_MLC_EMULATION)) {
-		pr_err("ubi: refuse attaching mtd%d - MLC NAND is not supported\n",
-			mtd->index);
-		return -EINVAL;
-	}
-
 	/* UBI cannot work on flashes with zero erasesize. */
 	if (!mtd->erasesize) {
 		pr_err("ubi: refuse attaching mtd%d - zero erasesize flash is not supported\n",
